### ISSUE TITLE
TextReader.NewLine

### ISSUE THREAD

**Issue by TonyValenti** [dotnet/corefx#39024]
Hi All,
TextWriter has a NewLine string that lets me specify what characters are output at the end of a line.
TextReader has no such property.  Can you please add one?

**Comment by vcsjones** [comment]
`TextReader` already handles `\n` and `\r\n` line endings, so in most cases it will do the "right" thing.

https://github.com/dotnet/corefx/blob/59d4a3ed40125b377083b10c8cb58d5f48522c0a/src/Common/src/CoreLib/System/IO/TextReader.cs#L186-L193

Are you asking for an API property that lets you be explicit about which line ending you prefer instead of the automatic handling?

**Comment by TonyValenti** [comment]
Yes.  I'm working with an interesting file format where line endings are []\r\n  (open bracket, close bracket, \r\n).  I'd really like to be able to just use TextReader to read from it.

**Comment by carlossanlop** [comment]
Thanks for your suggestion, @TonyValenti . Can you please write an official API Proposal? The detailed instructions can be found here: [Review Process](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/api-review-process.md)

Summarized steps:
1. Copy your original issue description and paste it in a comment in this issue, to preserve the original request.
2. Edit your issue description to match the formatting expected for an API Proposal: Rationale, Usage, Proposed API, Details, Open Questions, Pull Request, Updates. Here's a good example: https://github.com/dotnet/corefx/issues/271 . You can find many other examples across this repo (issues labeled `api-suggestion`).
3. When it's ready, ping the area owners (in this case, @JeremyKuhne and @carlossanlop). We'll be happy to review the proposal.

**Comment by carlossanlop** [comment]
Update:
Actually, I noticed we have this other `api-suggestion` in the `System.IO` area that asks for the same API but at a higher level (in StreamReader): https://github.com/dotnet/corefx/issues/29946

I think we should take the discussion for this API suggestion to that other thread instead. @JeremyKuhne thoughts?

**Comment by JeremyKuhne** [comment]
I think #29946 is asking for a return of what the line end was, not to specify the line end. I'd keep them linked, but separate for now.

Some thoughts looking at the code:

1. You can override and write your own `ReadLine()` logic (doesn't mean we can't do this, just a fact)
2. The current `ReadLine()` handles three different line endings (CR, LF, CR/LF)
3. How flexible should a new API be? One ending sequence? Multiple?
